tracer/src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -26,12 +24,17 @@ public static class SpanExtensions
         /// <param name="span">A span that belongs to the trace.</param>
         /// <param name="samplingPriority">The new sampling priority for the trace.</param>
         public static void SetTraceSamplingPriority(this ISpan span, SamplingPriority samplingPriority)
+        {
+            span.SetTraceSamplingPriority((int)samplingPriority);
+        }
+
+        internal static void SetTraceSamplingPriority(this ISpan span, int samplingPriority)
```

Comment:
Just thinking out loud - do we want this to be `public` if it's going to become the "preferred" way to set sampling priority? I'm undecided 🙂 and there's an easy path forward for users anyway so maybe easier not to

Reply:
I think we should improve the public method above instead of exposing this new one. There are two different cases where users could want to set sampling priority:
- User wants to manually sample or not sample a trace. That's the purposed of the above public extension method. However, `UserKeep(2)`/`UserReject(-1)` should be the only valid values, so there's room for improvement here 🧠🌩️:
  - accept a new public enum with only `UserKeep(2)`/`UserReject(-1)`
  - accept a Boolean value instead of an enum (true=keep, false=drop)
  - split into two methods: `Span.KeepTrace()`/`Span.DropTrace()`
  - whatever we decide, move the method to a newly exposed trace object (why are we setting sampling priority in a span when it's a trace-level property?)
- User is manually instrumenting a library we don't support OOTB and propagating sampling priority from an upstream service. They should use the sampling priority value as-is. Currently, this use-case is better served by setting the sampling priority on the `SpanContext` they create to be the "parent" of the local root span rather than using these extension methods.

Reply:
All this to say, "let's not" 😄